lib/jekyll/url.rb

```diff
@@ -67,32 +67,32 @@ def generate_url(template)
 
     def generate_url_from_hash(template)
       @placeholders.inject(template) do |result, token|
-        break result if result.index(':').nil?
+        break result if result.index(":").nil?
         if token.last.nil?
-          # Remove leading '/' to avoid generating urls with `//`
-          result.gsub(/\/:#{token.first}/, '')
+          # Remove leading "/" to avoid generating urls with `//`
+          result.gsub(%r!/:#{token.first}!, "")
         else
           result.gsub(/:#{token.first}/, self.class.escape_path(token.last))
         end
       end
     end
 
     def generate_url_from_drop(template)
-      template.gsub(/:([a-z_]+)/.freeze) do |match|
-        replacement = @placeholders.public_send(match.sub(':'.freeze, ''.freeze))
+      template.gsub(/:([a-z_]+)/) do |match|
+        replacement = @placeholders.public_send(match.sub(":".freeze, "".freeze))
         if replacement.nil?
-          ''.freeze
+          "".freeze
         else
           self.class.escape_path(replacement)
         end
-      end.gsub(/\/\//.freeze, '/'.freeze)
+      end.gsub(%r!//!, "/".freeze)
     end
 
     # Returns a sanitized String URL, stripping "../../" and multiples of "/",
     # as well as the beginning "/" so we can enforce and ensure it.
 
     def sanitize_url(str)
-      "/" + str.gsub(/\/{2,}/, "/").gsub(/\.+\/|\A\/+/, "")
+      "/" + str.gsub(%r!/{2,}!, "/").gsub(%r!\.+\/|\A\/+!, "")
```

Comment:
Good catch. Would it be worth adding a test for this method since nothing caught this?